core/shared/src/main/scala/zio/stm/TPriorityQueue.scala

```diff
@@ -22,26 +22,25 @@ import zio.stm.ZSTM.internal._
 import zio.{ Chunk, ChunkBuilder }
 
 /**
- * A simple `TPriorityQueue` implementation. A `TPriorityQueue` contains values
- * of type `A` that an `Ordering` is defined on. Unlike a `TQueue`, `take`
- * returns the highest priority value (the value that is first in the specified
- * ordering) as opposed to the first value offered to the queue. The ordering
- * that elements with the same priority will be taken from the queue is not
- * guaranteed.
+ * A `TPriorityQueue` contains values of type `A` that an `Ordering` is defined
+ * on. Unlike a `TQueue`, `take` returns the highest priority value (the value
+ * that is first in the specified ordering) as opposed to the first value
+ * offered to the queue. The ordering that elements with the same priority will
+ * be taken from the queue is not guaranteed.
  */
-final class TPriorityQueue[A] private (private val ref: TRef[SortedMap[A, Int]]) extends AnyVal {
+final class TPriorityQueue[A] private (private val ref: TRef[SortedMap[A, ::[A]]]) extends AnyVal {
 
   /**
    * Offers the specified value to the queue.
    */
   def offer(a: A): USTM[Unit] =
-    ref.update(map => map + (a -> map.get(a).fold(1)(_ + 1)))
+    ref.update(map => map + (a -> map.get(a).fold(::(a, Nil))(::(a, _))))
```

Comment:
we specify all the values at same partial ordering in a non-empty list ( a cons), as opposed to counting how many elements of same partial ordering we have.